include/curl/system.h

```diff
@@ -160,13 +160,19 @@
 
 #elif defined(__TANDEM)
 # if ! defined(__LP64)
-   /* Required for 32-bit NonStop builds only. */
 #  define CURL_TYPEOF_CURL_OFF_T     long long
 #  define CURL_FORMAT_CURL_OFF_T     "lld"
 #  define CURL_FORMAT_CURL_OFF_TU    "llu"
 #  define CURL_SUFFIX_CURL_OFF_T     LL
 #  define CURL_SUFFIX_CURL_OFF_TU    ULL
 #  define CURL_TYPEOF_CURL_SOCKLEN_T int
+# else
+#  define CURL_TYPEOF_CURL_OFF_T     long
+#  define CURL_FORMAT_CURL_OFF_T     "ld"
+#  define CURL_FORMAT_CURL_OFF_TU    "lu"
+#  define CURL_SUFFIX_CURL_OFF_T     L
+#  define CURL_SUFFIX_CURL_OFF_TU    UL
+#  define CURL_TYPEOF_CURL_SOCKLEN_T int
```

Comment:
Maybe move  the `CURL_TYPEOF_CURL_SOCKLEN_T` definition outside the #if/#else so that you don't have to duplicate it?

Reply:
> Maybe move the `CURL_TYPEOF_CURL_SOCKLEN_T` definition outside the #if/#else so that you don't have to duplicate it?

Which field is this used for? I actually have two different typedefs to work with. Not 100% sure int is correct anymore.

Reply:
This is better, and accurate, I think.

Reply:
> Aside, this really should be done by configure, no?

There's a major downside with that and the reason we moved away from doing it with configure: when you use the same headers for multiple builds/architectures.